src/pages/Stats.tsx

```diff
@@ -1,110 +1,112 @@
-import React, {useEffect} from 'react';
-import {makeStyles} from '@material-ui/core/styles';
+import React, {useEffect, useState} from 'react';
 import _ from 'lodash';
-import {fetchStats, AllStats} from '../api/stats';
+import {fetchStats} from '../api/stats';
+import {ListPuzzleStatsResponse} from '../shared/types';
+import {getUser} from '../store/user';
+import Flex from 'react-flexview';
+import {Helmet} from 'react-helmet';
+import Nav from '../components/common/Nav';
+import {formatMilliseconds} from '../components/Toolbar/Clock';
+import {makeStyles} from '@material-ui/core';
 
 const useStyles = makeStyles({
-  page: {
-    padding: 16,
-  },
-  liveContainer: {
-    padding: 16,
-  },
-  timeWindowContainer: {
-    padding: 16,
-  },
   header: {
-    fontSize: '200%',
-  },
-  counts: {
-    fontWeight: 'bold',
-  },
-  table: {
-    borderCollapse: 'collapse',
-    textAlign: 'right',
+    textAlign: 'center',
   },
 });
 
 const Stats: React.FC<{}> = () => {
+  const user = getUser();
   const classes = useStyles();
-  const [allStats, setAllStats] = React.useState({} as AllStats);
+
+  const [stats, setStats] = useState<ListPuzzleStatsResponse | null>(null);
+
   useEffect(() => {
-    async function refresh() {
-      const allStats = await fetchStats();
-      setAllStats(allStats);
-    }
-    refresh();
-  }, []);
+    user.listUserHistory().then((history: any) => {
+      console.log(`keys: ${_.keys(history)}`);
+      fetchStats({gids: _.keys(history)}).then((s) => {
```

Comment:
can we cap this at like, last 500 games? i'm not an expert in database performance but i'm worried that the backend could get slow if we try to do too many at once since the query involves joining with the game_events table (and there's no index on game event type afaik)